File: app/src/main/java/com/github/goldy1992/mp3player/dagger/modules/MediaControllerAdapterModule.java
package com.github.goldy1992.mp3player.dagger.modules;

import android.content.Context;

import com.github.goldy1992.mp3player.client.MediaControllerAdapter;
import com.github.goldy1992.mp3player.client.callbacks.MyMediaControllerCallback;
import com.github.goldy1992.mp3player.dagger.scopes.ComponentScope;

import dagger.Module;
import dagger.Provides;

@Module
public class MediaControllerAdapterModule {

    @ComponentScope
    @Provides
    MediaControllerAdapter provideMediaControllerAdapter(Context context, MyMediaControllerCallback myMediaControllerCallback) {
        return new MediaControllerAdapter(context, myMediaControllerCallback);
    }


}

File: client/src/main/java/com/github/goldy1992/mp3player/client/dagger/modules/MediaControllerAdapterModule.kt
package com.github.goldy1992.mp3player.client.dagger.modules

import android.content.Context
import android.support.v4.media.MediaBrowserCompat
import com.github.goldy1992.mp3player.client.AudioDataProcessor
import com.github.goldy1992.mp3player.client.MediaControllerAdapter
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ActivityRetainedComponent
import dagger.hilt.android.qualifiers.ApplicationContext
import dagger.hilt.android.scopes.ActivityRetainedScoped

@InstallIn(ActivityRetainedComponent::class)
@Module
class MediaControllerAdapterModule {

    @ActivityRetainedScoped
    @Provides
    fun providesMediaControllerAdapter(@ApplicationContext context: Context,
                                       mediaBrowserCompat: MediaBrowserCompat,
                                        audioDataProcessor : AudioDataProcessor)
            : MediaControllerAdapter {
        return MediaControllerAdapter(context, mediaBrowserCompat, audioDataProcessor)
    }
}